main.tf

```diff
@@ -29,11 +30,15 @@ resource "aws_grafana_workspace" "this" {
 ################################################################################
 
 locals {
+  create_role   = var.create && var.create_iam_role
   iam_role_name = coalesce(var.iam_role_name, var.name)
+
+  create_account_policy = local.create_role && var.account_access_type == "CURRENT_ACCOUNT"
+  create_custom_policy  = contains(var.data_sources, "CLOUDWATCH") || contains(var.data_sources, "AMAZON_OPENSEARCH_SERVICE") || contains(var.data_sources, "PROMETHEUS") || contains(var.notification_destinations, "SNS")
```

Comment:
```suggestion
  create_custom_policy  = length(setintersection(var.data_sources, ["CLOUDWATCH", "AMAZON_OPENSEARCH_SERVICE", "PROMETHEUS", "SNS"])) > 0
```

Reply:
I don't think we have ever used `setintersection()` in modules before.

Reply:
wow, thats much better and no I don't think we have used it yet - nice!